### ISSUE TITLE
Blocked Cases: Any Case Block In CG Blocks All Cases in CG from Eligible Case List

### ISSUE THREAD

**Issue by cholly75** [ustaxcourt/ef-cms#8777]
As a Case Services Supervisor, so that I do not incorrectly calendar cases in a partially eligible consolidated group, I need to block all cases in a group from being eligible for being calendared if any single case in the group is blocked.

Although Court process is to not calendar cases in a consolidated group when any of the cases are blocked, there is some confusion when looking at the Blocked Cases report in combination with the eligible case list for a given location due to the fact that the cases are split between the two displays when a group is only partially blocked.

We want to rationalize this process and consider all cases in a consolidated group as blocked when at least one of them is blocked. Additionally, we want to consolidated cases in the same group displayed together on the Blocked Cases report to facilitate ease of use.

## Acceptance Criteria
* If one or more cases in a consolidated group are blocked, all cases in the group display on the Blocked Cases report
* Consolidated cases displayed on the Blocked Cases report by virtue of being in a group with a blocked case display a reason of "Grouped with blocked case"
  - Cases that are blocked by virtue of being in a group with another blocked case display the red "Blocked" tag on the case header screen
* "Grouped with blocked case" added as a filter to the Reason dropdown menu
* Consolidated groups displayed on the Blocked Cases report are displayed together in the usual manner (Lead Case followed by member cases indented)
* When filters are applied, if any case displayed is in a consolidated group then the rest of the group is displayed grouped with that case
* If one or more cases in a consolidated group are blocked, none of the cases in the group display on the eligible case list for their location
* Add pagination to Blocked Cases report

## Story Definition of Ready (updated on 12/23/22)
The following criteria must be met in order for the user story to be picked up by the Flexion development team.
The user story must: 
- [ ] Is framed in business/user need, the value has been addressed.
- [ ] Includes acceptance criteria
- [ ] Has been refined
- [ ] Pre conditions have been satisfied. 

**Process:** 
Flexion developers and designers will test if the story meets acceptance criteria and test cases in Flexion dev and staging environments (“standard testing”). If additional acceptance criteria or testing scenarios are discovered while the story is in progress, a new story should be created, added to the backlog and prioritized by the product owner. 

## Definition of Done (Updated 5-19-22)
**Product Owner**
 - [ ]  Acceptance criteria have been met and validated on the Court's migration environment
 - [ ] Add scenario to testing document, if applicable (https://docs.google.com/spreadsheets/d/1FUHKC_YrT-PosaWD5gRVmsDzI1HS_U-8CyMIb-qX9EA/edit?usp=sharing)

**UX**
 - [ ] Business test scenarios have been refined to meet all acceptance criteria 
 - [ ] Usability has been validated
 - [ ] Wiki has been updated (if applicable) 
 - [ ] Story has been tested on a mobile device (for external users only)

**Engineering**
 - [ ] Automated test scripts have been written, including visual tests for newly added PDFs.
 - [ ] Field level and page level validation errors (front-end and server-side) integrated and functioning.
 - [ ] Verify that language for docket record for internal users and external users is identical.
 - [ ] New screens have been added to pa11y scripts.
 - [ ] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html.
 - [ ] READMEs, other appropriate docs, and swagger/APIs fully updated.
 - [ ] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required).
 - [ ] Interactors should validate entities before calling persistence methods.
 - [ ] Code refactored for clarity and to remove any known technical debt.
 - [ ] If new docket entries have been added as seed data to `efcms-local.json`, 3 local s3 files corresponding to that docketEntryId have been added to `web-api/storage/fixtures/s3/noop-documents-local-us-east-1`
 - [ ] Acceptance criteria for the story has been met.
 - [ ] If there are special instructions in order to deploy into the next environment, add them as a comment in the story.
 - [ ] If the work completed for the story requires a reindex without a migration, or any other special deploy steps, apply these changes to the following flexion branches:
   - [ ] experimental1
   - [ ] experimental2
   - [ ] experimental3
   - [ ] experimental4
   - [ ] experimental5
   - [ ] experimental6
   - [ ] develop
 - [ ] Reviewed by UX on a deployed environment.
 - [ ] Reviewed by PO on a deployed environment. Can be deployed to the Court's `test` environment if prod-like data is required. Otherwise deployed to any `experimental` environment.
 - [ ] Deployed to the Court's `staging` environment.

**Comment by katiecissell** [comment]
Pre-refinement notes:
- Reason filter in blocked cases report, how should it work? Might want a 'reason' for it.
- If we are treating the entire group as blocked, why not block them all when one is blocked?

**Comment by swongCO** [comment]
**UX Notes**
Link to [figma file](https://www.figma.com/design/Kr7UujumnSCYczXa2YLU3q/USTC-%7C-10440-Blocked-cases---consolidated-groups?node-id=39-7).

- [ ] Add new reason "Grouped with blocked case" to cases in a group that are blocked by association with a blocked case.
- [ ] Include the new reason as part of the Reason filter dropdown.
- [ ] Show the whole group in results when filters are applied if any filter is applicable to any of the cases.
- [ ] Pagination: Keep consolidated group together even if it means that there are fewer or more than 100 rows to prevent them from displaying on different pages.
- [ ] Add the red blocked tag to all cases in a group when one or more cases are blocked.
- [ ] If a case is severed from a blocked group and is not blocked otherwise, the blocked tag is removed.
- [ ] Each case in a consolidated group should count as one case in the table counter. For example, if there are 5 cases in a consolidated group, the table counter should reflect all 5 cases.

![Image](https://github.com/user-attachments/assets/f5e9565c-0237-4f88-a66a-1396bf6b7a5a)
![Image](https://github.com/user-attachments/assets/a830b26e-a945-4445-9d24-34918bd6b5ac)

**Comment by ttlenard** [comment]
## Test Cases

#### 1) Case Services Supervisor (CSS) runs a Blocked Cases Report for a Trial location; Lead case in a CG has a pending item; All cases in the group are displayed on the report and the reason is displayed correctly.

- As a Case Services Supervisor, navigate to a consolidated group of cases that have a case status of At Issue - Ready for Trial
     - Take note of the docket numbers of these cases 
     - Take note of the Requested place of trial
- Check the cases to see if any of them have a pending item, deadline or manual block that would cause the case(s) to be blocked.
- Ideally, you'll want no blockers on any cases to start with (suggest removing any pending items/deadlines if there are any to start fresh)
- Navigate to the Lead case in the group
- Click on the edit link for any document on the docket record
- Check the box for "Add to pending report"
- Click Save
- Be sure to note the docket number of the lead case. 
- There should now be a pending item in the tracked items tab for the docket entry that you just edited.
- Click on the Reports dropdown menu
- Select Blocked Cases
- Select the Trial Location for the Cases in the CG

Expected Results:
- [x] The lead case that you added the pending item is displayed in the list of blocked cases for the trial session. The Reason is "pending item"
- [x] The lead case has the Lead case icon next to the docket number
- [x] **NEW FUNCTIONALITY** - Each member case in the group is listed below the lead case and is indented slightly.  They have the member case icons.
- [x] **NEW FUNCTIONALITY** - The reason for each member case is "Grouped with blocked case"
- [x] **NEW FUNCTIONALITY** - After applying filters, the Lead case and member cases all display as a group 
     - [x] Filtering by Reason of "Pending item" displays the member cases that are with the Lead case (member case reasons say "Grouped with blocked case", even with the "Pending item" selected for reason.
     - [x] Filtering by Reason of "Grouped with blocked case" displays the lead case that is with the member cases (lead case reason still says "Pending item" even with the "Grouped with blocked case" selected as the reason
     - [x] Filtering by a reason other than the two above results in the consolidated case NOT displaying.  
- [x] **NEW FUNCTIONALITY** - The Lead case and all of the member cases have the "blocked" label when viewing the case header
- [x] Only the Lead case has anything in the Tracked items/Pending Items tab on the case. The member cases do not have any Pending items displayed on the case.



#### 2) Case Services Supervisor adds a new trial session; None of the cases in the consolidated group of cases from test case 1 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click the Add Trial Session
- Fill out the required fields for the trial session, making sure that you select hybrid + the location is the requested place of trial from the consolidated group in test case 1.
- Click save
- Navigate back to the trial session after you save it and review the Eligible Cases list

Expected Results:
- [x] **NEW FUNCTIONALITY** - None of the consolidated cases that are part of the group from test case 1 are displayed in the Eligible case list



#### 3) Case Services Supervisor removes the pending item on the blocked case from test case 1; Blocked label on all cases in the group are removed.

- Continuing from the previous test
- Navigate back to the lead case in the consolidated group from test case 1
- Click on the tracked items tab
- Click on the Pending Items tab
- Click Remove to remove the pending item that you added

Expected Results:
- [x] The Pending item is removed
- [x] The blocked label in the header is no longer displayed
- [x] **NEW FUNCTIONALITY** - The blocked label in all of the member cases is no longer present
- [x] The consolidated case group is no longer displayed when running a blocked cases report for the requested trial location



#### 4) Case Services Supervisor navigates to the new trial session from test case 2; Consolidated cases from test case 1 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click on the New tab
- Click on the City/State link for the Trial session you created in test case 2

Expected Results
- [x] All of the consolidated cases that are part of the group from test case 1 are displayed in the Eligible case list



#### 5) Case Services Supervisor (CSS) runs a Blocked Cases Report for a Trial location; Member case in a CG has a pending item; All cases in the group are displayed on the report and the reason is displayed correctly.

 As a Case Services Supervisor, navigate to a different consolidated group of cases that have a case status of At Issue - Ready for Trial
     - Take note of the docket numbers of these cases 
     - Take note of the Requested place of trial
- Check the cases to see if any of them have a pending item, deadline or manual block that would cause the case(s) to be blocked.
- Ideally, you'll want no blockers on any cases to start with (suggest removing any pending items/deadlines if there are any to start fresh)
- Navigate to a member case in the group
- Click on the edit link for any document on the docket record
- Check the box for "Add to pending report"
- Click Save
- Be sure to note the docket number of the member case. 
- There should now be a pending item in the tracked items tab for the docket entry that you just edited.
- Click on the Reports dropdown menu
- Select Blocked Cases
- Select the Trial Location for the Cases in the CG

Expected Results:
- [x] The member case that you added the pending item is displayed in the list of blocked cases for the trial session. The Reason is "pending item"
- [x] The member case has the member case icon next to the docket number
- [x] **NEW FUNCTIONALITY** - Each member case in the group is listed below the lead case and is indented slightly.  They have the member case icons.
- [x] **NEW FUNCTIONALITY** - The reason for the lead case and all the other member cases is "Grouped with blocked case"
- [x] **NEW FUNCTIONALITY** - After applying filters, the Lead case and member cases all display as a group 
     - [x] Filtering by Reason of "Pending item" displays all of the cases that are in the group (the member case that you added the pending item to says Pending item, the other cases say "Grouped with blocked case", even with the "Pending item" selected for reason.
     - [x] Filtering by Reason of "Grouped with blocked case" still displays the all of the cases in the group (the member case that you added the pending item to says Pending item, the other cases say "Grouped with blocked case" selected as the reason
     - [x] Filtering by a reason other than the two above results in the consolidated case NOT displaying.  
- [x] **NEW FUNCTIONALITY** - The member case that you selected and all of the other cases in the group has the "blocked" label when viewing the case header
- [x] Only the Member case that you selected has anything in the Tracked items/Pending Items tab on the case. The other cases in the group do not have any Pending items displayed on the case.


#### 6) Case Services Supervisor adds a new trial session; None of the cases in the consolidated group of cases from test case 5 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click the Add Trial Session
- Fill out the required fields for the trial session, making sure that you select hybrid + the location is the requested place of trial from the consolidated group in test case 5.
- Click save
- Navigate back to the trial session after you save it and review the Eligible Cases list

Expected Results:
- [x] **NEW FUNCTIONALITY** - None of the consolidated cases that are part of the group from test case 5 are displayed in the Eligible case list



#### 7) Case Services Supervisor removes the pending item on the blocked case from test case 5; Blocked label on all cases in the group are removed.

- Continuing from the previous test
- Navigate back to the member case in the consolidated group from test case 5
- Click on the tracked items tab
- Click on the Pending Items tab
- Click Remove to remove the pending item that you added

Expected Results:
- [x] The Pending item is removed
- [x] The blocked label in the header is no longer displayed
- [x] **NEW FUNCTIONALITY** - The blocked label in all of the member cases is no longer present
- [x] The consolidated case group is no longer displayed when running a blocked cases report for the requested trial location



#### 8) Case Services Supervisor navigates to the new trial session from test case 6; Consolidated cases from test case 5 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click on the New tab
- Click on the City/State link for the Trial session you created in test case 6

Expected Results
- [x] All of the consolidated cases that are part of the group from test case 5 are displayed in the Eligible case list



#### 9) Case Services Supervisor (CSS) runs a Blocked Cases Report for a Trial location; Lead case in a CG has a deadline; All cases in the group are displayed on the report and the reason is displayed correctly.

- As a Case Services Supervisor, navigate to a consolidated group of cases that have a case status of At Issue - Ready for Trial
     - Take note of the docket numbers of these cases 
     - Take note of the Requested place of trial
- Check the cases to see if any of them have a pending item, deadline or manual block that would cause the case(s) to be blocked.
- Ideally, you'll want no blockers on any cases to start with (suggest removing any pending items/deadlines if there are any to start fresh)
- Next, add in a deadline to the lead case in the group (do not add the deadline to any of the other cases), and be sure to track which docket number you just added it to.
- Navigate to the Lead case in the group
- Click on the create button
- Select Deadline
- Add a new deadline to the case
- Click Save
- Be sure to note the docket number of the lead case. 
- There should now be a deadline in the tracked items tab for the lead case.
- Click on the Reports dropdown menu
- Select Blocked Cases
- Select the Trial Location for the Cases in the CG

Expected Results:
- [x] The lead case that you added the deadline to is displayed in the list of blocked cases for the trial session. The Reason is "Due Date."
- [x] The lead case has the Lead case icon next to the docket number
- [x] **NEW FUNCTIONALITY** - Each member case in the group is listed below the lead case and is indented slightly.  They have the member case icons.
- [x] **NEW FUNCTIONALITY** - The reason for each member case is "Grouped with blocked case"
- [x] **NEW FUNCTIONALITY** - After applying filters, the Lead case and member cases all display as a group 
     - [x] Filtering by Reason of "Due Date" displays the member cases that are with the Lead case (member case reasons say "Grouped with blocked case", even with the "Due Date" selected for reason.
     - [x] Filtering by Reason of "Grouped with blocked case" displays the lead case that is with the member cases (lead case reason still says "Due Date" even with the "Grouped with blocked case" selected as the reason
     - [x] Filtering by a reason other than the two above results in the consolidated case NOT displaying.  
- [x] **NEW FUNCTIONALITY** - The Lead case and all of the member cases have the "blocked" label when viewing the case header
- [x] Only the Lead case has anything in the Tracked items/Deadlines tab on the case. The member cases do not have any Tracked items/deadlines displayed on the case.



#### 10) Case Services Supervisor adds a new trial session; None of the cases in the consolidated group of cases from test case 9 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click the Add Trial Session
- Fill out the required fields for the trial session, making sure that you select hybrid + the location is the requested place of trial from the consolidated group in test case 9.
- Click save
- Navigate back to the trial session after you save it and review the Eligible Cases list

Expected Results:
- [x] **NEW FUNCTIONALITY** - None of the consolidated cases that are part of the group from test case 9 are displayed in the Eligible case list



#### 11) Case Services Supervisor removes the Deadline on the blocked case from test case 9; Blocked label on all cases in the group are removed.

- Continuing from the previous test
- Navigate back to the lead case in the consolidated group from test case 9
- Click on the tracked items tab
- Click on the Deadlines tab
- Click Delete to remove the deadline that you added

Expected Results:
- [x] The Deadline is removed
- [x] The blocked label in the header is no longer displayed
- [x] **NEW FUNCTIONALITY** - The blocked label in all of the member cases is no longer present
- [x] The consolidated case group is no longer displayed when running a blocked cases report for the requested trial location



#### 12) Case Services Supervisor navigates to the new trial session from test case 10; Consolidated cases from test case 9 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click on the New tab
- Click on the City/State link for the Trial session you created in test case 10

Expected Results
- [x] All of the consolidated cases that are part of the group from test case 9 are displayed in the Eligible case list



#### 13) Case Services Supervisor (CSS) runs a Blocked Cases Report for a Trial location; Member case in a CG has a Deadline; All cases in the group are displayed on the report and the reason is displayed correctly.

- As a Case Services Supervisor, navigate to a consolidated group of cases that have a case status of At Issue - Ready for Trial
     - Take note of the docket numbers of these cases 
     - Take note of the Requested place of trial
- Check the cases to see if any of them have a pending item, deadline or manual block that would cause the case(s) to be blocked.
- Ideally, you'll want no blockers on any cases to start with (suggest removing any pending items/deadlines if there are any to start fresh)
- Next, add in a deadline to a member case in the group (do not add the deadline to any of the other cases), and be sure to track which docket number you just added it to.
- Navigate to a member case in the group
- Click on the create button
- Select Deadline
- Add a new deadline to the case
- Click Save
- Be sure to note the docket number of the member case. 
- There should now be a deadline in the tracked items tab for the member case.
- Click on the Reports dropdown menu
- Select Blocked Cases
- Select the Trial Location for the Cases in the CG

Expected Results:
- [x] The member case that you added the deadline to is displayed in the list of blocked cases for the trial session. The Reason is "Due Date."
- [x] The member case has the member case icon next to the docket number
- [x] **NEW FUNCTIONALITY** - Each member case in the group is listed below the lead case and is indented slightly.  They have the member case icons.
- [x] **NEW FUNCTIONALITY** - The reason for the member case is "Due Date", all other cases have the reason "Grouped with blocked case"
- [x] **NEW FUNCTIONALITY** - After applying filters, the Lead case and member cases all display as a group 
     - [x] Filtering by Reason of "Due Date" displays the member cases that are with the Lead case (member case reason says Deadline, and all other cases say "Grouped with blocked case", even with the "Due Date" selected for reason.
     - [x] Filtering by Reason of "Grouped with blocked case" displays the all of the cases that is with the member case (member case reason still says "Due Date" even with the "Grouped with blocked case" selected as the reason
     - [x] Filtering by a reason other than the two above results in the consolidated case NOT displaying.  
- [x] **NEW FUNCTIONALITY** - All of the cases in the group have the "blocked" label when viewing the case header
- [x] Only the member case has anything in the Tracked items/Deadlines tab on the case. The other cases do not have any Tracked items/deadlines displayed on the case.


#### 14) Case Services Supervisor adds a new trial session; None of the cases in the consolidated group of cases from test case 13 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click the Add Trial Session
- Fill out the required fields for the trial session, making sure that you select hybrid + the location is the requested place of trial from the consolidated group in test case 13.
- Click save
- Navigate back to the trial session after you save it and review the Eligible Cases list

Expected Results:
- [x] **NEW FUNCTIONALITY** - None of the consolidated cases that are part of the group from test case 13 are displayed in the Eligible case list



#### 15) Case Services Supervisor removes the Deadline on the blocked case from test case 13; Blocked label on all cases in the group are removed.

- Continuing from the previous test
- Navigate back to the member case in the consolidated group from test case 13
- Click on the tracked items tab
- Click on the Deadlines tab
- Click Delete to remove the deadline that you added

Expected Results:
- [x] The Deadline is removed
- [x] The blocked label in the header is no longer displayed
- [x] **NEW FUNCTIONALITY** - The blocked label in all of the cases is no longer present
- [x] The consolidated case group is no longer displayed when running a blocked cases report for the requested trial location



#### 16) Case Services Supervisor navigates to the new trial session from test case 14; Consolidated cases from test case 13 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click on the New tab
- Click on the City/State link for the Trial session you created in test case 14

Expected Results
- [x] All of the consolidated cases that are part of the group from test case 13 are displayed in the Eligible case list



#### 17) Case Services Supervisor (CSS) runs a Blocked Cases Report for a Trial location; Lead case and at least one member case in a CG has a pending item and a deadline; All cases in the group are displayed on the report and the reason is displayed correctly.

- As a Case Services Supervisor, navigate to a consolidated group of cases that have a case status of At Issue - Ready for Trial
     - Take note of the docket numbers of these cases 
     - Take note of the Requested place of trial
- Check the cases to see if any of them have a pending item, deadline or manual block that would cause the case(s) to be blocked.
- Ideally, you'll want no blockers on any cases to start with (suggest removing any pending items/deadlines if there are any to start fresh)
- Next, add in a deadline and a pending item to a some of the cases in the group (do not add the deadline and pending item to ALL of the other cases, just some), and be sure to track which docket numbers you just added it to, and which ones you didn't.
- Be sure to note the docket numbers of what you added for each case in the group. 
- Click on the Reports dropdown menu
- Select Blocked Cases
- Select the Trial Location for the Cases in the CG

Expected Results:
- [x] The case(s) that you added a deadline to is displayed in the list of blocked cases for the trial session. The Reason is "Due Date."
- [x] The case(s) that you added a pending item to is displayed in the list of blocked cases for the trial session. The Reason is "Pending Item."
- [x] The lead case has the Lead case icon next to the docket number
- [x] **NEW FUNCTIONALITY** - Each member case in the group is listed below the lead case and is indented slightly.  They have the member case icons.
- [x] **NEW FUNCTIONALITY** - The reason for each case that you didn't add any tracked items to is "Grouped with blocked case"
- [ ] **NEW FUNCTIONALITY** - After applying filters, the Lead case and member cases all display as a group 
     - [x] Filtering by Reason of "Due Date" displays the cases in the group (case that didn't get a tracked item have reasons that say "Grouped with blocked case", even with the "Due Date" selected for reason).
     - [x] Filtering by Reason of "Grouped with blocked case" displays cases in the group (case reasons of "Due Date" and "Pending item" are displayed even with the "Grouped with blocked case" selected as the reason).
     - [x] Filtering by a reason other than the two above results in the consolidated case NOT displaying.  
- [x] **NEW FUNCTIONALITY** - The Lead case and all of the member cases have the "blocked" label when viewing the case header
- [x] Only cases that you added tracked items have items in the Tracked items/Deadlines tab on the case. The other cases do not have any Tracked items/deadlines displayed on the case.


#### 18) Case Services Supervisor adds a new trial session; None of the cases in the consolidated group of cases from test case 17 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click the Add Trial Session
- Fill out the required fields for the trial session, making sure that you select hybrid + the location is the requested place of trial from the consolidated group in test case 17.
- Click save
- Navigate back to the trial session after you save it and review the Eligible Cases list

Expected Results:
- [x] **NEW FUNCTIONALITY** - None of the consolidated cases that are part of the group from test case 17 are displayed in the Eligible case list



#### 19) Case Services Supervisor removes the Deadline/pending items on the blocked cases from test case 17; Blocked label on all cases in the group are removed.

- Continuing from the previous test
- Navigate back to the cases in the consolidated group from test case 17
- Click on the tracked items tab for each case
- Click Delete to remove any deadline that you added and remove the pending item if you added one to a case (do this for each case)


Expected Results:
- [x] The Deadline is removed
- [x] The pending item is removed
- [x] The blocked label in the headers are no longer displayed
- [x] **NEW FUNCTIONALITY** - The blocked label in all of the cases is no longer present
- [x] The consolidated case group is no longer displayed when running a blocked cases report for the requested trial location



#### 20) Case Services Supervisor navigates to the new trial session from test case 18; Consolidated cases from test case 17 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click on the New tab
- Click on the City/State link for the Trial session you created in test case 18

Expected Results
- [x] All of the consolidated cases that are part of the group from test case 17 are displayed in the Eligible case list




#### 21) Case Services Supervisor (CSS) runs a Blocked Cases Report for a Trial location; Lead case in a CG has a manual block; All cases in the group are displayed on the report and the reason is displayed correctly.

- As a Case Services Supervisor, navigate to a consolidated group of cases that have a case status of At Issue - Ready for Trial
     - Take note of the docket numbers of these cases 
     - Take note of the Requested place of trial
- Check the cases to see if any of them have a pending item, deadline or manual block that would cause the case(s) to be blocked.
- Ideally, you'll want no blockers on any cases to start with (suggest removing any pending items/deadlines if there are any to start fresh)
- Next, add in a manual block to the lead case in the group (do not add the manual block to any of the other cases), and be sure to track which docket number you just added it to.
- Cick on the Case information tab
- Click on the link to "Add Manual Block"
- Input a reason as to why you are blocking the case
- Click Block Case
- Be sure to note the docket number of the lead case. 
- There should now be a Blocked label in the case header for the lead case, and there is manual block information on the Case information tab.
- Click on the Reports dropdown menu
- Select Blocked Cases
- Select the Trial Location for the Cases in the CG

Expected Results:
- [x] The lead case that you added the manual block to is displayed in the list of blocked cases for the trial session. The Reason is whatever you typed in for the reason.
- [x] The lead case has the Lead case icon next to the docket number
- [x] **NEW FUNCTIONALITY** - Each member case in the group is listed below the lead case and is indented slightly.  They have the member case icons.
- [x] **NEW FUNCTIONALITY** - The reason for each member case is "Grouped with blocked case"
- [x] **NEW FUNCTIONALITY** - After applying filters, the Lead case and member cases all display as a group 
     - [x] Filtering by Reason of "Manual Block" displays the member cases that are with the Lead case (member case reasons say "Grouped with blocked case", even with the "Manual Block" selected for reason.)
     - [x] Filtering by Reason of "Grouped with blocked case" displays the lead case that is with the member cases (lead case reason still says whatever you typed in for the reason even with the "Grouped with blocked case" selected as the reason)
     - [x] Filtering by a reason other than the two above results in the consolidated case NOT displaying.  
- [x] **NEW FUNCTIONALITY** - The Lead case and all of the member cases have the "blocked" label when viewing the case header
- [x] There is no data in the Tracked items/Deadlines tab on any of the cases. 



#### 22) Case Services Supervisor adds a new trial session; None of the cases in the consolidated group of cases from test case 21 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click the Add Trial Session
- Fill out the required fields for the trial session, making sure that you select hybrid + the location is the requested place of trial from the consolidated group in test case 21.
- Click save
- Navigate back to the trial session after you save it and review the Eligible Cases list

Expected Results:
- [x] **NEW FUNCTIONALITY** - None of the consolidated cases that are part of the group from test case 21 are displayed in the Eligible case list



#### 23) Case Services Supervisor removes the Manual Block on the blocked case from test case 21; Blocked label on all cases in the group are removed.

- Continuing from the previous test
- Navigate back to the lead case in the consolidated group from test case 21
- Click on the case information tab
- Click Remove Block and then Yes, Remove Block to remove the block that you added

Expected Results:
- [x] The Manual Block is removed
- [x] The blocked label in the header is no longer displayed
- [x] **NEW FUNCTIONALITY** - The blocked label in all of the member cases is no longer present
- [x] The consolidated case group is no longer displayed when running a blocked cases report for the requested trial location



#### 24) Case Services Supervisor navigates to the new trial session from test case 22; Consolidated cases from test case 21 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click on the New tab
- Click on the City/State link for the Trial session you created in test case 22

Expected Results
- [x] All of the consolidated cases that are part of the group from test case 21 are displayed in the Eligible case list



#### 25) Case Services Supervisor (CSS) runs a Blocked Cases Report for a Trial location; Member case in a CG has a manual block; All cases in the group are displayed on the report and the reason is displayed correctly.

- As a Case Services Supervisor, navigate to a consolidated group of cases that have a case status of At Issue - Ready for Trial
     - Take note of the docket numbers of these cases 
     - Take note of the Requested place of trial
- Check the cases to see if any of them have a pending item, deadline or manual block that would cause the case(s) to be blocked.
- Ideally, you'll want no blockers on any cases to start with (suggest removing any pending items/deadlines if there are any to start fresh)
- Next, add in a manual block to a member case in the group (do not add the manual block to any of the other cases), and be sure to track which docket number you just added it to.
- Cick on the Case information tab
- Click on the link to "Add Manual Block"
- Input a reason as to why you are blocking the case
- Click Block Case
- Be sure to note the docket number of the member case. 
- There should now be a Blocked label in the case header for the member case, and there is manual block information on the Case information tab.
- Click on the Reports dropdown menu
- Select Blocked Cases
- Select the Trial Location for the Cases in the CG

Expected Results:
- [x] The member case that you added the manual block to is displayed in the list of blocked cases for the trial session. The Reason is whatever you typed in for the reason.
- [x] The member case has the member case icon next to the docket number
- [x] **NEW FUNCTIONALITY** - Each member case in the group is listed below the lead case and is indented slightly.  They have the member case icons.
- [x] **NEW FUNCTIONALITY** - The reason for all of the other member cases and the lead case is "Grouped with blocked case"
- [x] **NEW FUNCTIONALITY** - After applying filters, the Lead case and member cases all display as a group 
     - [x] Filtering by Reason of "Manual Block" displays all of the cases in the group (other member cases and lead case reason says "Grouped with blocked case", even with the "Manual Block" selected for reason.)
     - [x] Filtering by Reason of "Grouped with blocked case" displays all of the cases in the group (Member case reason still says whatever you typed in for the reason even with the "Grouped with blocked case" selected as the reason)
     - [x] Filtering by a reason other than the two above results in the consolidated case NOT displaying.  
- [x] **NEW FUNCTIONALITY** - The Lead case and all of the member cases have the "blocked" label when viewing the case header
- [x] There is no data in the Tracked items/Deadlines tab on any of the cases. 


#### 26) Case Services Supervisor adds a new trial session; None of the cases in the consolidated group of cases from test case 25 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click the Add Trial Session
- Fill out the required fields for the trial session, making sure that you select hybrid + the location is the requested place of trial from the consolidated group in test case 25.
- Click save
- Navigate back to the trial session after you save it and review the Eligible Cases list

Expected Results:
- [x] **NEW FUNCTIONALITY** - None of the consolidated cases that are part of the group from test case 25 are displayed in the Eligible case list



#### 27) Case Services Supervisor removes the Manual Block on the blocked case from test case 25; Blocked label on all cases in the group are removed.

- Continuing from the previous test
- Navigate back to the member case in the consolidated group from test case 25
- Click on the case information tab
- Click Remove Block and then Yes, Remove Block to remove the block that you added

Expected Results:
- [x] The Manual Block is removed
- [x] The blocked label in the header is no longer displayed
- [x] **NEW FUNCTIONALITY** - The blocked label in all of the cases in the group is no longer present
- [x] The consolidated case group is no longer displayed when running a blocked cases report for the requested trial location



#### 28) Case Services Supervisor navigates to the new trial session from test case 26; Consolidated cases from test case 25 display in the Eligible case list.

- Continuing from the previous test
- Click on Trial Sessions
- Click on the New tab
- Click on the City/State link for the Trial session you created in test case 26

Expected Results
- [x] All of the consolidated cases that are part of the group from test case 25 are displayed in the Eligible case list


#### 29) New Paginator is added to the Blocked Cases Report; Consolidated Groups displayed on the Blocked case report do not get split up with pagination.

- Navigate to the Blocked Cases report
- Select a city that will have many blocked cases (LA, New York, etc.)

Expected results:
- [x] **NEW FUNCTIONALITY** - There is now a paginator present when there are over 100 blocked cases for a selected city
- [x] Paginator not present for cities with less than 100 results
- [x] **NEW FUNCTIONALITY** - Paginator not present if selected filters result in less than 100 results
- [x] **NEW FUNCTIONALITY** - None of the blocked consolidated cases are split up onto 2 different pages

**Comment by zachrog** [comment]
Questions for Chris:
1: On the case Detials screen if a member case is blocked through association what should be displayed inside case information blocked area.
Do not change the manual block section.

2: If a case is blocked, and then removed from a consolidated Group does it still retain its block?
yes

3: If a case is blocked by association, and then removed from a consolidated group does it still retain its block?
No

4: If a case is blocked by association, and then the blocked case is removed from the group is the original case still blocked?
No

4: When unsetAsHighPriority of a case do we need to make sure all consolidated cases are not visible on the eligible cases screen?
When changing unsetting the priority of a case do not worry about updating or checking all in consolidated group. No one uses this feature.

**Comment by zachrog** [comment]
Chris H questions

- What is a crazy number of Blocked cases that can exist? 1000
- How many cases should exist on a page? 100

I typically make our app support 10x requirements so supporting 10000 blocked cases on the report.

**Comment by zachrog** [comment]
@ttlenard 

It would be a good idea to test the trial session planning report because the way we count blocked cases has changed. Now in the trial session planning report if there is a consolidated group with 2 cases and one of them has a block on it the planning report will (and should according to Chris H.) show 2 cases. Previously it would show 1.

**Comment by katiecissell** [comment]
UX Feedback: 

- [ ] Paginator should be above and below the results list. Its currently only displaying below.
- [ ] Date blocked should display for all cases in the group (currently not displaying for the ones grouped with a blocked case)

**Comment by katiecissell** [comment]
We are moving the location above the paginator/count to resolve some overlapping issues. So the location will look more like a header now.

**Comment by ttlenard** [comment]
@pavanyenduri @zachrog 

This story is blocked from testing until Bug ustaxcourt/ef-cms#8921 is resolved.  Happy to chat about it if needed.
